transport/udp/udp.go

```diff
@@ -0,0 +1,390 @@
+				mangos.PropRemoteAddr: conn.RemoteAddr(),
+			},
+		},
+		readBuf: make([]byte, udpMaxBuf),
+	}, nil
+}
+
+/*
+ * Listener methods
+ */
+type listener struct {
+	*udpEndpoint
+	*net.UDPConn // implements listener.Close()
+}
+
+func (l *listener) Address() string {
+	return "udp://" + l.addr.String()
+}
+
+func (l *listener) Listen() (err error) {
+	l.UDPConn, err = net.ListenUDP("udp", l.addr)
+	return
+}
+
+// halfPipePeers records remote endpoints that have connected to this Listener
+var halfPipePeers = struct {
+	// which peer connected the listening endpoint
+	byAddr map[string]*udpHalfPipe
+	sync.Mutex
+}{byAddr: make(map[string]*udpHalfPipe)}
+
+// Accept sorts messages into the mailboxes of existing peers, until a new
+// peer connects; causing a new udpHalfPipe to be returned.
+func (l *listener) Accept() (mangos.Pipe, error) {
+	for {
+		var buf = make([]byte, udpMaxBuf)
+
+		n, addr, err := l.UDPConn.ReadFromUDP(buf)
+		if err != nil {
+			return nil, err
+		}
+		msg, err := l.extractMsgFromPacket(buf[:n])
+		if err != nil {
+			return nil, fmt.Errorf("invalid packet from %s: %s", addr, err)
+		}
+
+		halfPipePeers.Lock()
+		uh, peerExists := halfPipePeers.byAddr[addr.String()]
+		if !peerExists {
+			uh = &udpHalfPipe{
+				udpPipe: &udpPipe{
+					udpEndpoint: udpEndpoint{
+						addr:  l.addr,
+						opts:  l.opts,
+						proto: l.proto,
+					},
+					UDPConn: l.UDPConn,
+					open:    true,
+					props: map[string]interface{}{
+						mangos.PropLocalAddr:  l.addr,
+						mangos.PropRemoteAddr: addr,
+					},
+				},
+				peer:    addr,
+				inqueue: make(chan *mangos.Message, udpMaxBacklog),
+			}
+			halfPipePeers.byAddr[addr.String()] = uh
+		}
+		halfPipePeers.Unlock()
+
+		uh.inqueue <- msg
+
+		if !peerExists {
+			return uh, nil
+		}
+	}
+	return nil, mangos.ErrClosed
+}
+
+// udpTran implements the Transport interface for udp.
+type udpTran struct{}
+
+func (t *udpTran) Scheme() string {
+	return "udp"
+}
+
+func (t *udpTran) NewDialer(addr string, sock mangos.Socket) (mangos.PipeDialer, error) {
+	ep, err := t.newEndpoint(addr, sock)
+	if err != nil {
+		return nil, err
+	}
+	return &dialer{ep}, nil
+}
+
+func (t *udpTran) NewListener(addr string, sock mangos.Socket) (mangos.PipeListener, error) {
+	ep, err := t.newEndpoint(addr, sock)
+	if err != nil {
+		return nil, err
+	}
+	return &listener{udpEndpoint: ep}, nil
+}
+
+func (t *udpTran) newEndpoint(addr string, sock mangos.Socket) (*udpEndpoint, error) {
+	var e = &udpEndpoint{
+		proto: sock.GetProtocol(),
+		opts:  make(map[string]interface{}),
+	}
+	var err error
+
+	if addr, err = mangos.StripScheme(t, addr); err != nil {
+		return nil, err
+	}
+	if e.addr, err = net.ResolveUDPAddr("udp", addr); err != nil {
+		return nil, err
+	}
+
+	if val, err := sock.GetOption(mangos.OptionMaxRecvSize); err == nil {
+		if maxrx, ok := val.(int); ok && maxrx < udpMaxBuf {
+			return nil, fmt.Errorf("MaxRecvSize(%d) to small for udp - need %d", maxrx, udpMaxBuf)
```

Comment:
But that doesn't seem to be what the code does, if I'm reading it.  It seems like you don't let the user set a smaller maximum receive size.  I can see perfectly valid reasons to limit the max recv size even smaller than the maximum UDP message size.  Especially this is true for jumbo UDP.

Perhaps you mean the sense of the < sign to be reversed?